### PR TITLE
Migrate Materialize command to vtctldclient

### PR DESCRIPTION
**Pull request by mattlord**
## Description

[`vtctldclient` is replacing the legacy `vtctlclient`](https://github.com/vitessio/vitess/pull/12866). As part of that effort this PR migrates the [`Materialize`](https://vitess.io/docs/reference/vreplication/materialize/) command to vtctldclient.

Here's a manual test case that demonstrates the usage:
```
cd examples/local
./101_initial_cluster.sh; mysql < ../common/insert_commerce_data.sql

vtctldclient --server localhost:15999 materialize --workflow product_sales --target-keyspace commerce create --source-keyspace commerce --table-settings '[{"target_table": "sales_by_sku", "create_ddl": "create table sales_by_sku (sku varbinary(128) not null primary key, orders bigint, revenue bigint)", "source_expression": "select sku, count(*) as orders, sum(price) as revenue from corder group by sku"}]' --cells zone1 --cells zone2 --tablet-types replica

vtctldclient --server localhost:15999 materialize --workflow product_sales --target-keyspace commerce show

mysql --binary-as-hex=false commerce -e "select * from corder; select * from sales_by_sku"

vtctldclient --server localhost:15999 materialize --workflow product_sales --target-keyspace commerce cancel
```

With the output being:
```
Materialization workflow product_sales successfully created in the commerce keyspace. Use show to view the status.

{
  "workflows": [
    {
      "name": "product_sales",
...

+----------+-------------+----------+-------+
| order_id | customer_id | sku      | price |
+----------+-------------+----------+-------+
|        1 |           1 | SKU-1001 |   100 |
|        2 |           2 | SKU-1002 |    30 |
|        3 |           3 | SKU-1002 |    30 |
|        4 |           4 | SKU-1002 |    30 |
|        5 |           5 | SKU-1002 |    30 |
+----------+-------------+----------+-------+
+----------+--------+---------+
| sku      | orders | revenue |
+----------+--------+---------+
| SKU-1001 |      1 |     100 |
| SKU-1002 |      4 |     120 |
+----------+--------+---------+

Successfully cancelled the product_sales workflow in the commerce keyspace
```

## Related Issue(s)

 - Fixes: https://github.com/vitessio/vitess/issues/13968
 - Fixes: https://github.com/vitessio/vitess/issues/8845

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/cmd/vtctldclient/command/vreplication/common/utils.go [hunk 1]
```diff
@@ -185,8 +185,8 @@ func AddCommonCreateFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&CreateOptions.TabletTypesInPreferenceOrder, "tablet-types-in-preference-order", true, "When performing source tablet selection, look for candidates in the type order as they are listed in the tablet-types flag.")
 	cmd.Flags().StringVar(&CreateOptions.OnDDL, "on-ddl", onDDLDefault, "What to do when DDL is encountered in the VReplication stream. Possible values are IGNORE, STOP, EXEC, and EXEC_IGNORE.")
 	cmd.Flags().BoolVar(&CreateOptions.DeferSecondaryKeys, "defer-secondary-keys", false, "Defer secondary index creation for a table until after it has been copied.")
-	cmd.Flags().BoolVar(&CreateOptions.AutoStart, "auto-start", true, "Start the MoveTables workflow after creating it.")
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/vreplication/common/utils.go:188]: Up to you (I think it's fine how you have it!): could also pass a string arg to `AddCommonFlags` and start doing stuff like:

```go
cmd.Flags().BoolVar(&CreateOptions.AutoStart, "auto-start", true, fmt.Sprintf("Start the %s workflow after creating it.", operation))
```

File: go/cmd/vtctldclient/command/vreplication/materialize/create.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+var (
+	createOptions = struct {
+		SourceKeyspace string
+		TableSettings  tableSettings
+	}{}
+
+	// create makes a MaterializeCreate gRPC call to a vtctld.
+	create = &cobra.Command{
+		Use:                   "create",
+		Short:                 "Create and run a Materialize VReplication workflow.",
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/vreplication/materialize/create.go:43]: TIOLI: might be worth expanding on the syntax for `--table-settings` in a `Long` text (essentially to replace/generate [this](https://vitess.io/docs/19.0/reference/vreplication/materialize/#json-spec-details)

**Review by ajm188 — APPROVED** [review]
looks good! just some minor suggestions

File: go/cmd/vtctldclient/command/vreplication/materialize/create.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		Example:               `vtctldclient --server localhost:15999 materialize --workflow product_sales --target-keyspace commerce create --source-keyspace commerce --table-settings '[{"target_table": "sales_by_sku", "create_ddl": "create table sales_by_sku (sku varbinary(128) not null primary key, orders bigint, revenue bigint)", "source_expression": "select sku, count(*) as orders, sum(price) as revenue from corder group by sku"}]' --cells zone1 --cells zone2 --tablet-types replicas`,
```
Comment (rohit-nayak-ps) [inline go/cmd/vtctldclient/command/vreplication/materialize/create.go:44]: should be  => `--tablet-types replica`

**Review by rohit-nayak-ps — APPROVED** [review]

File: go/cmd/vtctldclient/command/vreplication/materialize/create.go (hunk 3 above)
Comment (mattlord) [reply go/cmd/vtctldclient/command/vreplication/materialize/create.go:44]: Good catch! I'll fix that (was testing the validation with that).

File: go/cmd/vtctldclient/command/vreplication/materialize/create.go (hunk 2 above)
Comment (mattlord) [reply go/cmd/vtctldclient/command/vreplication/materialize/create.go:43]: Good idea! I'll do that.

**Pull request merged**